scripts/new-version-pr-generator.js

```diff
@@ -0,0 +1,146 @@
+
+See what’s new: https://github.com/Shopify/polaris-react/releases/tag/${version}
+
+cc @Shopify/polaris-reviewers
+
+---
+
+<details>
+<summary>🚨 What to do if you see “Your tests failed on CircleCI”?</summary>
+
+If tests fail, you may have to troubleshoot the problem locally.
+
+1. Checkout the \`update-polaris-${version}\` branch:
+    \`\`\`bash
+    dev cd ${repository}
+    git fetch
+    git checkout update-polaris-${version}
+    dev up
+    \`\`\`
+1. Apply changes/fixes
+1. Commit
+1. Push your work to the \`update-polaris-${version}\` branch:
+    \`\`\`bash
+    git push origin update-polaris-${version}
+    \`\`\`
+1. Repeat until the build & tests go ✅
+1. 🎩
+1. :shipit:
+</details>
+`.trim();
+}
+
+function failedUpdateMessage(repository, version) {
+  return `
+The automatic branch creation for ${repository} failed. This can be due to many reasons. To resolve this follow these steps:
+
+  1. Checkout "${repository}"
+  2. Get the latest version of master "git checkout master && git pull"
+  3. Create a new branch based on the version that failed "git checkout -b update-polaris-v${version}"
+  4. Update "@shopify/polaris" by running "npx yarn upgrade @shopify/polaris@${version}"
+  5. Add the changed files to git by running "git add package.json yarn.lock"
+  6. Commit the changes with "git commit -m 'Update @shopify/polaris to ${version}'"
+  7. Push the changes with "git push origin update-polaris-v${version}"
+  8. Create a new pull request for the branch in "${repository}"
+`;
+}
+
+mkdir(sandbox);
+
+repositories.forEach((repository) => {
+  try {
+    const repositoryDirectory = resolve(sandbox, repository);
+
+    // Clone into the sandbox directory
+    retry(
+      execSync(
+        `git clone --branch ${baseBranch} --single-branch https://${polarisBotToken}@github.com/Shopify/${repository}.git`,
+        {
+          cwd: sandbox,
+          stdio: 'inherit',
+        },
+      ),
+    );
+
+    const commands = [
+      `git checkout -b update-polaris-${releaseVersion}`,
+      `npx yarn@${YARN_VERSION} upgrade @shopify/polaris@${releaseVersion.replace(
+        'v',
+        '',
+      )} --no-progress --ignore-engines`,
+      `git add package.json yarn.lock`,
+      `${shopifyPolarisBotGitOverride} git commit -m "Update @shopify/polaris to ${releaseVersion}" --author "${polarisBotName} <${polarisBotEmail}>" -m "${releaseVersion}" --allow-empty`,
+      `git push origin update-polaris-${releaseVersion}`,
+      `curl -d '${JSON.stringify({
+        title: `Update @shopify/polaris to ${releaseVersion} 🚀`,
+        body: pullRequestTemplate(repository, releaseVersion),
+        head: `update-polaris-${releaseVersion}`,
+        base: baseBranch,
+      })}' -X POST https://api.github.com/repos/shopify/${repository}/pulls?access_token=${polarisBotToken}`,
+    ];
+
+    // Run the commands in the cloned repository directories
+    commands.forEach((command) => {
```

Comment:
Are you sure this works? I'm pretty sure this `forEach` is queueing up all your commands at the same time. See the following console output:
```
arr = [1,2,3,4,5,6,7,8,9,10]
(10) [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
const func = async (number) => { console.log(number); const ms = Math.random()*500; await new Promise(resolve => setTimeout(resolve, ms)); console.log(ms); console.log(number) }
undefined
arr.forEach(func)
1
2
3
4
5
6
7
8
9
10

50.21044903148086
9
61.93249680065127
6
79.9692515536532
2
80.7784522197541
8
111.60789086028467
5
154.02466156309458
3
191.62890656360963
1
297.53635281697353
10
368.0370214847132
7
388.1897108923006
4
```

Reply:
This is great thanks @amrocha

Reply:
@amrocha I have approached this slightly different to how we discussed but I think it resolves the asynchronous problems you raised. Let me know if you have any feedback.